Content.Server/_DV/CosmicCult/MonumentSystem.cs

```diff
@@ -0,0 +1,369 @@
+    {
+        _audio.PlayEntity(_audio.ResolveSound(monument.Comp.InfusionSFX), cultist, monument);
+        _popup.PopupEntity(Loc.GetString("cosmiccult-entropy-inserted", ("count", cultist.Comp.EntropyStored)), cultist, cultist);
+        monument.Comp.TotalEntropy += cultist.Comp.EntropyStored;
+        cultist.Comp.EntropyStored = 0;
+        Dirty(cultist, cultist.Comp);
+        _cosmicRule.UpdateCultData(monument);
+        return true;
+    }
+
+    /// <summary>
+    /// Method for adding itemized Entropy to The Monument.
+    /// </summary>
+    private bool AddEntropy(Entity<MonumentComponent> monument, EntityUid entropy, Entity<CosmicCultComponent> cultist)
+    {
+        var quant = TryComp<StackComponent>(entropy, out var stackComp) ? stackComp.Count : 1;
+        monument.Comp.TotalEntropy += quant;
+        cultist.Comp.EntropyBudget += quant;
+
+        Dirty(cultist, cultist.Comp);
+        _cosmicRule.UpdateCultData(monument);
+
+        _popup.PopupEntity(Loc.GetString("cosmiccult-entropy-inserted", ("count", quant)), cultist, cultist);
+        _audio.PlayEntity(_audio.ResolveSound(monument.Comp.InfusionSFX), cultist, monument);
+        QueueDel(entropy);
+        return true;
+    }
+
+        public void UpdateMonumentAppearance(Entity<MonumentComponent> ent, bool tierUp) // this is kinda awful, but it works, and i've seen worse. improve it at thine leisure
+    {
+        if (_cosmicRule.AssociatedGamerule(ent) is not {} cult)
+            return;
+        if (!TryComp<CosmicFinaleComponent>(ent, out var finaleComp))
+            return;
+        _appearance.SetData(ent, MonumentVisuals.Monument, cult.Comp.CurrentTier);
+
+        switch (cult.Comp.CurrentTier)
+        {
+            case 3:
+                _appearance.SetData(ent, MonumentVisuals.Tier3, true);
+                break;
+            case 2:
+                _appearance.SetData(ent, MonumentVisuals.Tier3, false);
+                break;
+        }
+
+        if (tierUp)
+        {
+            var transformComp = EnsureComp<MonumentTransformingComponent>(ent);
+            transformComp.EndTime = _timing.CurTime + ent.Comp.TransformTime;
+            _appearance.SetData(ent, MonumentVisuals.Transforming, true);
+        }
+
+        if (finaleComp.CurrentState != FinaleState.Unavailable)
+            _appearance.SetData(ent, MonumentVisuals.FinaleReached, true);
+    }
+
+    //note - these ar the thresholds for moving to the next tier
+    //so t1 -> 2 needs 20% of the crew
+    //t2 -> 3 needs 40%
+    //and t3 -> finale needs an extra 20 entropy
+    public void UpdateMonumentReqsForTier(Entity<MonumentComponent> monument, int tier)
+    {
+        if (_cosmicRule.AssociatedGamerule(monument) is not {} cult)
+            return;
+
+        var tier3NumCrew = Math.Round((double)cult.Comp.TotalCrew / 100 * _config.GetCVar(DCCVars.CosmicCultTargetConversionPercent)); // 40% of current pop
```

Comment:
Give this a better name, you're not paid by the byte. (it's kinda cursed that something named "num" is a double instead of an int but it's fine)
Also remove comment since that only holds if the value is not changed.

```suggestion
        var numCrewForTier3 = Math.Round((double)cult.Comp.TotalCrew / 100 * _config.GetCVar(DCCVars.CosmicCultTargetConversionPercent));
```